crates/lexgen/benches/benchmarks.rs

```diff
@@ -5,9 +5,9 @@ use criterion::{black_box, criterion_group, criterion_main, Criterion};
 
 #[inline(never)]
 fn lex_lua(s: &str) {
-    let mut lexer = Lexer::new(s);
-    while let Some(next) = lexer.next() {
-        next;
+    let lexer = Lexer::new(s);
+    for res in lexer {
+        assert!(res.is_ok());
```

Comment:
This now includes `is_ok` and whatever `assert!` expands to in the benchmarked thing. Since we only compare these results with a previous iteration of the same benchmark that may be OK, but I wonder if this will make the benchmark more noisy, or the `assert!` and `is_ok` overhead will be too much compared to the actual thing we want to benchmark.

Does it work if you do `for res in lexer {}`?

Reply:
`for _ in lexer {}` still gives sensible `cargo bench` output (slightly better than before)